packages/components/src/TimeInput.test.tsx

```diff
@@ -149,6 +151,64 @@ describe('select and type', () => {
 
     testSelectAndType(1, '000000', '00:00:00');
   });
+  it('handles backspace', () => {
+    // Replace selected section with fixed-width spaces
+    testSelectAndType(
+      0,
+      '{backspace}',
+      `${FIXED_WIDTH_SPACE}${FIXED_WIDTH_SPACE}:34:56`
+    );
+    testSelectAndType(
+      3,
+      '{backspace}',
+      `12:${FIXED_WIDTH_SPACE}${FIXED_WIDTH_SPACE}:56`
+    );
+
+    // Allow deleting digits from the end
+    testSelectAndType(9, '{backspace}', `12:34`);
+
+    // Add missing mask chars
+    testSelectAndType(9, '{backspace}{backspace}12', `12:31:2`);
+  });
+
+  it('trims trailing mask and spaces', () => {
+    const { unmount } = makeTimeInput();
+    const input: HTMLInputElement = screen.getByRole('textbox');
+
+    input.setSelectionRange(3, 3);
+
+    userEvent.type(input, '{backspace}');
+
+    input.setSelectionRange(9, 9);
+
+    userEvent.type(input, '{backspace}');
+
+    expect(input.value).toEqual(`12`);
+
+    input.setSelectionRange(1, 1);
+
+    userEvent.type(input, '{backspace}');
+
+    expect(input.value).toEqual(``);
+
+    unmount();
+  });
+
+  it('existing invalid behaviors that might need to be fixed', () => {
+    // Expected: '20:34:56'?
+    testSelectAndType(1, '5{arrowleft}2', `25:34:56`);
+
+    // Fill in with zeros when skipping positions. Expected: '03:34:56'
+    testSelectAndType(0, '{backspace}3', `${FIXED_WIDTH_SPACE}3:34:56`);
+
+    // Not sure it's ok to skip to the next section when the input isn't valid for the current section
+    // Expected: '03:34:56'?
+    testSelectAndType(0, '35', `03:54:56`);
```

Comment:
This should be `03:54:56`, this is correct. Yes it should jump to the next section if the previous section is complete, which it is in this case (since you put `3`, which can only go in the second spot, finishing off that segment).